examples/browser/test/pages/main-page.ts

```diff
@@ -62,10 +69,102 @@ export class MainPage {
     }
 
     public getxBarTabPosition(tabNumber: Number): Number {
-        return browser.getLocation(`ul.p-MenuBar-content > .p-MenuBar-item:nth-child(${tabNumber}`, 'x')
+        return browser.getLocation(`ul.p-MenuBar-content > .p-MenuBar-item:nth-child(${tabNumber}`, 'x');
     }
 
     public getxMenuPosition(): Number {
-        return this.driver.getLocation('.p-Widget.p-Menu.p-MenuBar-menu', 'x')
+        return this.driver.getLocation('.p-Widget.p-Menu.p-MenuBar-menu', 'x');
+    }
+
+    public isFolderStructureExpandable(): Boolean {
+        browser.waitForExist('.theia-TreeNodeCaption');
+        return browser.isExisting('span.theia-ExpansionToggle');
+    }
+
+    public isRightClickDirectoryAvailable(): Boolean {
+        console.log('*****************************' + browser.isExisting('li.p-TabBar-tab'));
+        browser.click('li.p-TabBar-tab');
+        browser.rightClick('#files.p-Widget.theia-Tree.theia-FileTree.theia-Files.p-StackedPanel-child', 50, 50);
+        return browser.isVisible('.p-Widget.p-Menu');
+    }
+
+    public isRightClickInMainPanelAvailable(): Boolean {
+        browser.rightClick('#theia-main-dock-panel');
+        return browser.isVisible('.p-Widget.p-Menu');
+    }
+
+    public isDoubleClickOnFileAvailable(): Boolean {
+        browser.waitForExist('.theia-TreeNodeCaption');
+        browser.doubleClick('.theia-TreeNodeCaption');
+        return browser.isVisible('#theia-left-stack.p-Widget.p-Panel.p-StackedPanel.p-SplitPanel-child');
+    }
+
+    /*open is at position 1 in the context menu*/
+    public isContextMenuOpenAvailable(filePositionNumber: Number): Boolean {
+        browser.waitForVisible(`#files div.theia-TreeNode.theia-FileStatNode:nth-child(${filePositionNumber}`);
+        browser.rightClick(`#files div.theia-TreeNode.theia-FileStatNode:nth-child(${filePositionNumber})`, 0, 0);
+        browser.waitForExist('div.p-Widget.p-Menu ul.p-Menu-content');
+        browser.click('ul li.p-Menu-item:nth-child(1)');
+        return browser.isExisting('ul.p-TabBar-content');
+    }
+
+    /*open with is at position 2 in the context menu*/
+    public isContextMenuOpenWithAvailable(filePositionNumber: Number): Boolean {
```

Comment:
These positions could change... 
I don't think it's a good idea to hardcode that...
Would it be possible to check if it's there at any position ?